docs/canonical-ways-to-investigate-bugs.md

```diff
@@ -0,0 +1,368 @@
+
+Problems can appear at the beginning of the Send flow, typically when inputting the transaction data in the send form.
+
+We have in live-common a "TransactionStatus" concept which virtually everything is good to broadcast the transaction. The error is likely to be contained in that.
+
+An easy way to try that is to use `ledger-live getTransactionStatus`. The parameters it takes correspond to the Send form fields. Each family can add more field parameter (see the `families/*/cli-transaction.js` definition or `ledger-live help`)
+
+It also takes the same parameter of a sync, so an --xpub can be used (if not provided, we use a connected Ledger device):
+
+```
+ledger-live getTransactionStatus -c ZEN --xpub xpub6DBXGC1faXnWSWofjAXjfxihNmYEcJj7KcWytuc7AZK39zS8KdgbosCjnPyHRbwKxcJxnWbzLwZYaPHFE3zoQZmGDbPzwRuoSfFaeSTVwUT --recipient znaE1JH6YjjWiG23bqdCkBiC4KwZ9vo7hB6 --amount 100
+```
+
+In that case, I don't have enough ZEN to spend, so like the recipient field will show an error, it will also show me in the CLI:
+
+```
+{
+  status: '{"errors":{"amount":"{\\"name\\":\\"NotEnoughBalance\\",\\"message\\":\\"NotEnoughBalance\\",\\"stack\\":\\"Error\\\\n    at new CustomError (/Users/grenaudeau/dev/ledger-live-common/cli/node_modules/@ledgerhq/errors/lib/helpers.js:27:18)\\\\n    at remapLibcoreErrors (/Users/grenaudeau/dev/ledger-live-common/cli/node_modules/@ledgerhq/live-common/lib/libcore/platforms/nodejs.js:543:16)\\\\n    at remapLibcoreErrors (/Users/grenaudeau/dev/ledger-live-common/cli/node_modules/@ledgerhq/live-common/lib/libcore/errors.js:18:10)\\\\n    at /Users/grenaudeau/dev/ledger-live-common/cli/node_modules/@ledgerhq/live-common/lib/libcore/getFeesForTransaction.js:46:42\\"}"},"warnings":{},"estimatedFees":"0","amount":"10000000000","totalSpent":"10000000000"}',
+  transaction: '{"amount":"10000000000","recipient":"znaE1JH6YjjWiG23bqdCkBiC4KwZ9vo7hB6","useAllAmount":false,"family":"bitcoin","feePerByte":"1","networkInfo":{"family":"bitcoin","feeItems":{"items":[{"key":"0","speed":"high","feePerByte":"10"},{"key":"1","speed":"standard","feePerByte":"10"},{"key":"2","speed":"low","feePerByte":"10"}],"defaultFeePerByte":"10"}}}'
+}
+```
+
+But if all is good, I will have some data:
+
+```
+{
+  status: '{"errors":{},"warnings":{},"estimatedFees":"226","amount":"10000000","totalSpent":"10000226"}',
+  transaction: '{"amount":"10000000","recipient":"znaE1JH6YjjWiG23bqdCkBiC4KwZ9vo7hB6","useAllAmount":false,"family":"bitcoin","feePerByte":"1","networkInfo":{"family":"bitcoin","feeItems":{"items":[{"key":"0","speed":"high","feePerByte":"10"},{"key":"1","speed":"standard","feePerByte":"10"},{"key":"2","speed":"low","feePerByte":"10"}],"defaultFeePerByte":"10"}}}'
+}
+```
+
+Just using `getTransactionStatus` will show you if a problem occur typically related to fetching fees or any other problem related to building a transaction. Now if the problem occurs at the device time, this will not be enough...
```

Comment:
```suggestion
Just using `getTransactionStatus` will show you if a problem occurs related to fetching fees or any other problem related to building a transaction. Now if the problem occurs at the device time, this will not be enough...
```